### ISSUE TITLE
Mismatch between processor architectures "mscorlib.dll", "AMD64" (Code=MSB3270)

### ISSUE THREAD

**Issue by lackovic** [ionide/ionide-vscode-fsharp#289]
I followed "Quick Install Guide - Windows" on http://ionide.io , without using  chocolatey, downloaded from Microsoft and installed F# 4.0 for Windows adn Microsoft Build Tools 2015.

Any valid F# code gets underlined with a green squiggle. When I pass with the mouse cursor over it I get the following message:

"There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "mscorlib.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. (Code=MSB3270)."

**Comment by lackovic** [comment]
Updated Visual Studio Code to version 1.7.2 and the problem disappeared.

**Comment by Krzysztof-Cieslak** [comment]
CC: @dsyme

Don, any idea why FCS would return something like that?

**Comment by dsyme** [comment]
@Krzysztof-Cieslak It's an MSBuild message - either from project cracker of from the FCS use of MSBuild.  Strange. Really need someone to debug into it or we need reliable repro steps that can be used on any machine?

**Comment by Splike** [comment]
I experienced the same thing today on my Windows 10 machine.
Only happens on .fs files and not .fsi or .fsx

Steps to reproduce:
- install [mono](http://www.mono-project.com/) (This step might not be necesarry, but I had it installed before anything else, so I can't say if it has any influence or not to the error)
- Install VS code
- Install extension "ionide-fsharp"
- Follow the "Quick Install Guide - Windows" on http://ionide.io
  - install [Visual F# Tools 4.0 RTM](https://www.microsoft.com/en-us/download/details.aspx?id=48179)
  - install [Microsoft Build Tools 2015](https://www.microsoft.com/en-us/download/details.aspx?id=48159&wa=wsignin1.0)
- create .fs file with any code or comment on the very first line and the error will appear.

**Comment by lilred** [comment]
If it's any use, we've had this bug appear on multiple students' machines. I think they were all Windows machines but I can't recall for sure.

**Comment by Krzysztof-Cieslak** [comment]
Can anyone check if it's still problem on the latest?

**Comment by ramoncreager** [comment]
I just experienced this on Windows 7, all patches up to date. Simple "Hello World" program in Atom using Ionide, which I just installed. (printfn "Hello World" is the only line.) Tried leaving the first line blank, no dice. I had already coincidentally installed Visual Studio Community 2015.


PS C:\Users\Ramon> fsc
Microsoft (R) F# Compiler version 14.0.23020.0
Copyright (c) Microsoft Corporation. All Rights Reserved.
